src/test/app/Delegate_test.cpp

```
@@ -1271,6 +1389,74 @@ class Delegate_test : public beast::unit_test::suite
+
+            // alice gives granular permission to bob of MPTokenIssuanceLock
+            env(delegate::set(alice, bob, {"MPTokenIssuanceLock"}));
+            env.close();
+            mpt.set(
+                {.account = alice,
+                 .flags = tfMPTLock | tfFullyCanonicalSig,
+                 .delegate = bob});
+        }
+
+        // tfInnerBatchTxn won't block delegated transaction
+        {
+            Env env(*this);
+            Account alice{"alice"};
+            Account bob{"bob"};
+            env.fund(XRP(100000), alice, bob);
+            env.close();
+
+            auto const mptID = makeMptID(env.seq(alice), alice);
+            MPTTester mpt(env, alice, {.fund = false});
+            env.close();
+            mpt.create({.flags = tfMPTCanLock});
+            env.close();
+
+            // alice gives granular permission to bob of MPTokenIssuanceLock
+            env(delegate::set(
+                alice, bob, {"MPTokenIssuanceLock", "MPTokenIssuanceUnlock"}));
+            env.close();
+
+            auto const seq = env.seq(alice);
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+
+            Json::Value jv1;
+            jv1[sfTransactionType] = jss::MPTokenIssuanceSet;
+            jv1[sfAccount] = alice.human();
+            jv1[sfDelegate] = bob.human();
+            jv1[sfSequence] = seq + 1;
+            jv1[sfMPTokenIssuanceID] = to_string(mptID);
+            jv1[sfFlags] = tfMPTLock;
+
+            Json::Value jv2;
+            jv2[sfTransactionType] = jss::MPTokenIssuanceSet;
+            jv2[sfAccount] = alice.human();
+            jv2[sfDelegate] = bob.human();
+            jv2[sfSequence] = seq + 2;
+            jv2[sfMPTokenIssuanceID] = to_string(mptID);
+            jv2[sfFlags] = tfMPTUnlock;
+
+            // batch::inner will set tfInnerBatchTxn, this should not
+            // block delegated transaction
+            env(batch::outer(alice, seq, batchFee, tfAllOrNothing),
```

This test doesn't actually tell you if the inner transactions failed or not. If you're adding batch transaction tests then I would add them to the Batch_test.cpp or we should make public the testing helpers so that you can accurately test the result of the inner batch transactions.

Here are some Batch_test.cpp Delegate tests: https://github.com/XRPLF/rippled/blob/7e24adbdd0b61fb50967c4c6d4b27cc6d81b33f3/src/test/app/Batch_test.cpp#L3665

---

Since the more detailed tests are already in Batch_test, this test case is just to make sure the permission related masks are accurate in `TxFlags.h`. 

For example, in `TxFlags.h`, if `tfMPTokenIssuanceSetPermissionMask = ~(tfUniversal | tfMPTLock | tfMPTUnlock);`. This case will pass.
If `tfMPTokenIssuanceSetPermissionMask = ~(tfMPTLock | tfMPTUnlock);` (missing tfUniversal), it will error `tecNO_DELEGATE_PERMISSION`

---

The InnerResult of this test is a `tecNO_DELEGATE_PERMISSION` if you do not include the tfUniversal flag, however the test does silently fail....

---

but I tested by removing tfUniversal flag, the test will report:
#137 failed: apply: Got tecNO_DELEGATE_PERMISSION (Delegated account lacks permission to perform this transaction.); Expected tesSUCCESS (The transaction was applied. Only final in a validated ledger.)
It's not failing silently.

---

It is. The error you reference above is from a different test in your test case. If the batch failed it wouldn't fail like that. That is a failure from a env.require and comes from line `1409: mpt.set(`

Comment out the `// tfFullyCanonicalSig won't block delegated transaction` test and you will see the batch test is silently failing.

---

yeah, I'll create another PR to modify

---

@Bronek @dangell7 I just created the PR https://github.com/XRPLF/rippled/pull/5476